examples/translation_example.py

```diff
@@ -0,0 +1,463 @@
+                            i += 1
+                    else:
+                        str += ch
+                        i += 1
+                if str != "":
+                    tokenized_snt_lst.append(str)
+                # Add the tokenized sentences to the list
+                tokenized_sentences.extend(tokenized_snt_lst)
+            else:
+                tokenized_sentences.append(line)
+
+        tokenized_sentence_len = []
+        for sentence in tokenized_sentences:
+            sent = sentence.split()
+            # removing the sentences with word length greater than threshold as the model may not be able translate it due to constraint on output token size
+            if len(sent) <= self.translation_config.max_words_per_sen:
+                tokenized_sentence_len.append(sentence)
+            else:
+                pass
+        return tokenized_sentence_len
+
+    def process_input_text(self, ddf):
+        ddf = ddf.to_pandas()
+        ddf["indic_proc_text"] = ddf["text"].apply(self.custom_tokenize)
+        ddf["doc_id"] = range(1, len(ddf) + 1)
+        ddf = ddf.explode("indic_proc_text")
+        ddf = ddf.reset_index(drop=True)
+        ddf = ddf.drop("text", axis=1)
+        ddf[["start_sym", "indic_proc_text", "end_sym"]] = ddf["indic_proc_text"].apply(
+            self.truncate_start_end_symbols
+        )
+        ddf = cudf.DataFrame.from_pandas(ddf)
+        return ddf
+
+    def concat_strings(self, series):
+        return "".join(series)
+
+    def combine_text(self, ddf):
+        engligh_stop_flag = ddf["indic_proc_text"].str.endswith(".")
+        hindi_stop_flag = ddf["translation"].str.endswith("|")
+        ddf["translation"][~engligh_stop_flag & hindi_stop_flag] = ddf[
+            "translation"
+        ].str.rstrip("|")
+        ddf["translation"] = (
+            ddf["start_sym"] + ddf["translation"].str.strip() + ddf["end_sym"]
+        )
+        ddf["indic_proc_text"] = (
+            ddf["start_sym"] + ddf["indic_proc_text"] + ddf["end_sym"]
+        )
+        return ddf
+
+    def grouping(self, ddf):
+        ddf = ddf.to_pandas()
+        agg_funcs = {
+            "translation": self.concat_strings,
+            "indic_proc_text": self.concat_strings,
+        }
+        other_columns = {
+            col: "first"
+            for col in ddf.columns
+            if col not in agg_funcs and col != "doc_id"
+        }
+
+        agg_funcs.update(other_columns)
+        ddf = ddf.groupby("doc_id").agg(agg_funcs).reset_index()
+        ddf = cudf.DataFrame.from_pandas(ddf)
+        return ddf
+
+    def combine_text_false(self, ddf):
+        ddf["translation"] = ddf["start_sym"] + ddf["indic_proc_text"] + ddf["end_sym"]
+        ddf["indic_proc_text"] = ddf["translation"]
+        return ddf
+
+    def atleast_letter(self, df):
+        df = df.to_pandas()
+        df["isalpha"] = df["indic_proc_text"].apply(self.has_alphabet_characters)
+        df = cudf.DataFrame(df)
```

Comment:
Do not think you need a round-trip here to cuDF

```suggestion
        df["isalpha"] = df["indic_proc_text"].str.contains('[a-zA-Z]')
```

Reply:
This only passes for english letters. Standalone has check with isalpha() which also allows unicode letters to pass. That's why I need to do this.

Reply:
Cool